source/slang/reflection.cpp

```diff
@@ -860,6 +860,24 @@ SLANG_API int spReflectionTypeLayout_getGenericParamIndex(SlangReflectionTypeLay
     }
 }
 
+SLANG_API SlangReflectionTypeLayout* spReflectionTypeLayout_getPendingDataTypeLayout(SlangReflectionTypeLayout* inTypeLayout)
+{
+    auto typeLayout = convert(inTypeLayout);
+    if(!typeLayout) return nullptr;
+
+    auto pendingDataTypeLayout = typeLayout->pendingDataTypeLayout.Ptr();
+    return convert(pendingDataTypeLayout);
+}
+
+SLANG_API SlangReflectionVariableLayout* spReflectionVariableLayout_getPendingDataLayout(SlangReflectionVariableLayout* inVarLayout)
+{
+    auto varLayout = convert(inVarLayout);
+    if(!varLayout) return nullptr;
+
+    auto pendingDataLayout = varLayout->pendingVarLayout.Ptr();
+    return convert(pendingDataLayout);
```

Comment:
The 'convert' method name seems like it might do work/construct something. Perhaps a name that reflects it's casty/no work nature would be better like 'asExternal'.

Reply:
Yeah, `asExternal` would be more explicit, but then we'd need `asInternal` to go in the other direction, and most of this is just "I have this variable that has the wrong type and I just want it in the *other* type."

A better solution in the long run is to carefully set it up so that the Slang implementation code actually defines exactly the types forward-declared in the header. That would simplify this kind of logic, and would also give us the big benefit that the public API types would start to show something useful in a debugger, so long as debug symbols for Slang are linked in.

Reply:
For what it's worth - I think using `asExternal` and `asInternal` would be reasonable way to go.